VSDiagnostics/VSDiagnostics/VSDiagnostics/Utilities/Extensions.cs

```diff
@@ -144,5 +145,37 @@ public static bool IsAsync(this IMethodSymbol methodSymbol)
                 || methodSymbol.ReturnType.MetadataName == typeof(Task).Name
                 || methodSymbol.ReturnType.MetadataName == typeof(Task<>).Name;
         }
+
+        public static bool IsDefinedInAncestor(this IMethodSymbol methodSymbol)
+        {
+            var containingType = methodSymbol?.ContainingType;
+            if (containingType == null)
+            {
+                return false;
+            }
+
+            var interfaces = containingType.AllInterfaces;
+            foreach (var @interface in interfaces)
+            {
+                var interfaceMethods = @interface.GetMembers().Select(containingType.FindImplementationForInterfaceMember);
+                if (interfaceMethods.Any(method => method.Equals(methodSymbol)))
+                {
+                    return true;
+                }
+            }
+
+            var baseType = containingType.BaseType;
+            while (baseType != null)
```

Comment:
Indeed, that's very weird... Let me know if I can do anything there to help.

When this PR is merged, I'll update the other PR to make use of this, and the other extension method, and remove all the superfluous tests.

> ... thanks for staying with me!

No probs ;) I had the intention to do some regular contribution to this lib, but I'm quite occupied with some other stuff, so it might take a while before I'll be able to put in some new analyzers :(